Build failure for wasm32-wasip1 target

```
   Compiling diffutils v0.5.0 (/home/runner/work/diffutils/diffutils)
error[E0433]: failed to resolve: could not find `unix` in `os`
  --> src/cmp.rs:18:14
   |
18 | use std::os::unix::fs::MetadataExt;
   |              ^^^^ could not find `unix` in `os`
   |
note: found an item that was configured out
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/mod.rs:29:5
   |
   = note: the item is gated here
note: found an item that was configured out
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/mod.rs:84:41
   |
   = note: the item is gated here
error[E0599]: no method named `dev` found for struct `Metadata` in the current scope
  --> src/cmp.rs:55:30
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.ino() == dev_null.ino();
   |                              ^^^ method not found in `Metadata`
   |
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:406:8
   |
   = note: the method is available for `Metadata` here
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `dev` is implemented but not in scope; perhaps you want to import it
   |
 6 + use std::os::wasi::fs::MetadataExt;
   |
error[E0599]: no method named `dev` found for struct `Metadata` in the current scope
  --> src/cmp.rs:55:48
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.ino() == dev_null.ino();
   |                                                ^^^ method not found in `Metadata`
   |
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:406:8
   |
   = note: the method is available for `Metadata` here
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `dev` is implemented but not in scope; perhaps you want to import it
   |
 6 + use std::os::wasi::fs::MetadataExt;
   |
error[E0599]: no method named `ino` found for struct `Metadata` in the current scope
  --> src/cmp.rs:55:64
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.ino() == dev_null.ino();
   |                                                                ^^^
   |
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:408:8
   |
   = note: the method is available for `Metadata` here
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `ino` is implemented but not in scope; perhaps you want to import it
   |
 6 + use std::os::wasi::fs::MetadataExt;
   |
help: there is a method `into` with a similar name
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.into() == dev_null.ino();
   |                                                                  +
error[E0599]: no method named `ino` found for struct `Metadata` in the current scope
  --> src/cmp.rs:55:82
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.ino() == dev_null.ino();
   |                                                                                  ^^^
   |
  --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:408:8
   |
   = note: the method is available for `Metadata` here
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `ino` is implemented but not in scope; perhaps you want to import it
   |
 6 + use std::os::wasi::fs::MetadataExt;
   |
help: there is a method `into` with a similar name
   |
55 |     let is_dev_null = stdout.dev() == dev_null.dev() && stdout.ino() == dev_null.into();
   |                                                                                    +
error[E0599]: no method named `size` found for struct `Metadata` in the current scope
   --> src/cmp.rs:333:40
    |
333 |         let (a_size, b_size) = (a_meta.size(), b_meta.size());
    |                                        ^^^^ method not found in `Metadata`
    |
   --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:412:8
    |
    = note: the method is available for `Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `size` is implemented but not in scope; perhaps you want to import it
    |
  6 + use std::os::wasi::fs::MetadataExt;
    |
error[E0599]: no method named `size` found for struct `Metadata` in the current scope
   --> src/cmp.rs:333:55
    |
333 |         let (a_size, b_size) = (a_meta.size(), b_meta.size());
    |                                                       ^^^^ method not found in `Metadata`
    |
   --> /rustc/01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf/library/std/src/os/wasi/fs.rs:412:8
    |
    = note: the method is available for `Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `size` is implemented but not in scope; perhaps you want to import it
    |
  6 + use std::os::wasi::fs::MetadataExt;
    |
Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `diffutils` (lib) due to 7 previous errors
```

--- a/src/cmp.rs
+++ b/src/cmp.rs
@@ -11,10 +11,10 @@ use std::iter::Peekable;
 use std::process::ExitCode;
 use std::{cmp, fs, io};
 
-#[cfg(not(target_os = "windows"))]
+#[cfg(unix)]
 use std::os::fd::{AsRawFd, FromRawFd};
 
-#[cfg(not(target_os = "windows"))]
+#[cfg(unix)]
 use std::os::unix::fs::MetadataExt;
 
 #[cfg(target_os = "windows")]
@@ -38,7 +38,7 @@ fn usage_string(executable: &str) -> String {
     format!("Usage: {executable} <from> <to>")
 }
 
-#[cfg(not(target_os = "windows"))]
+#[cfg(unix)]
 fn is_stdout_dev_null() -> bool {
     let Ok(dev_null) = fs::metadata("/dev/null") else {
         return false;
@@ -58,6 +58,11 @@ fn is_stdout_dev_null() -> bool {
     std::mem::forget(stdout_file);
 
     is_dev_null
+}
+
+#[cfg(not(any(unix, target_os = "windows")))]
+fn is_stdout_dev_null() -> bool {
+    false
 }
 
 pub fn parse_params<I: Iterator<Item = OsString>>(mut opts: Peekable<I>) -> Result<Params, String> {
@@ -330,7 +335,7 @@ pub fn cmp(params: &Params) -> Result<Cmp, String> {
 
     if let (Ok(a_meta), Ok(b_meta)) = (fs::metadata(&params.from), fs::metadata(&params.to)) {
         #[cfg(not(target_os = "windows"))]
-        let (a_size, b_size) = (a_meta.size(), b_meta.size());
+        let (a_size, b_size) = (a_meta.len(), b_meta.len());
 
         #[cfg(target_os = "windows")]
         let (a_size, b_size) = (a_meta.file_size(), b_meta.file_size());

